src/amber/router/params_parser.cr

```diff
@@ -90,10 +91,10 @@ module Amber::Router
           case json = JSON.parse(body).raw
           when Hash
             json.each do |key, value|
-              params[key.as(String)] = value.to_s
+              params[key.as(String)] = value
```

Comment:
If JSONType is a string I think we should set it as a string otherwise just set it to value. WDYT?

Reply:
```cr
          when Hash
            json.each do |key, value|
              case value.raw.class
              when String
                params[key.as(String)] = value.to_s
              else
                params[key.as(String)] = value
              end
            end
          when Array
            params["_json"] = json.to_s
          end
```

Reply:
JSON::Parse returns always a  JSON::Any

Reply:
Yeah. That's why I'm saying that if it's a string instead of a hash we should just use the string value as the value instead of json::any. If it's anything except the string then just set it to the Json any value. See above.

Reply:
<img width="933" alt="screen shot 2017-10-25 at 11 11 13 am" src="https://user-images.githubusercontent.com/1685772/32006649-565c5224-b975-11e7-9e6f-8aee2028d4fe.png">

Reply:
How is value an Array(JSON::Type) for you? The array should still be JSON::Any. In my tests it is.

Reply:
We should pair on that one, to give you more context. But in a nutshell JSON.parse will return a JSON::Type and since we are not ‘unboxing’ to a specific type it reads as the first type `maybe` (I can be completely wrong here). 

This might work `value.as(String).raw.is_a? String` now this part of the params parser only gets trigger if the request is Application/JSON, this works fine as it is. So I think we can leave it

Reply:
`.raw` only exists on unions and abstracts though. So if it's an Array(JSON::Any) .raw won't exist. In my tests values from JSON.parse are always JSON::Any though. Let me know when you have time to pair on this. Hopefully we can get it merged today.

Reply:
Ok so I didn't know you could do parse_raw. I was basing this off of the old implementation. So this is what I should have said.

```cr
          when Hash
            json.each do |key, value|
             if value.is_a? String
                params[key.as(String)] = value.to_s
             else
                params[key.as(String)] = value
             end
            end
          when Array
            params["_json"] = json
          end
```